source/faq.txt

```diff
@@ -47,3 +47,69 @@ In MongoDB deployments with mixed versions of :binary:`~bin.mongod`, it is
 possible to get an ``Unrecognized pipeline stage name: '$sample'``
 error. To mitigate this situation, explicitly configure the partitioner
 to use and define the Schema when using DataFrames.
+
+How do I use MongoDB BSON types that are unsupported in Spark?
+--------------------------------------------------------------
+
+Some custom MongoDB BSON types, such as ``ObjectId``\s, are unsupported 
+in Spark.
+
+The MongoDB Spark Connector converts custom MongoDB data types to and 
+from extended JSON-like representations of those data types that are 
+compatible with Spark. See :ref:`<bson-spark-datatypes>` for a list of 
+custom MongoDB types and their Spark counterparts.
+
+Spark Datasets
+~~~~~~~~~~~~~~
+
+To create a standard Dataset with custom MongoDB data types, use 
+``fieldType`` helpers: 
+
+.. code-block:: scala
+   
+   import com.mongodb.spark.sql.fieldTypes
+ 
+   case class MyData(id: fieldTypes.ObjectId, a: Int)
+   val ds = spark.createDataset(Seq(MyData(fieldTypes.ObjectId(new ObjectId()), 99)))
+   ds.show()
+
+The example above creates a Dataset where:
+
+- The ``id`` field is a custom MongoDB BSON type, ``ObjectId``, defined 
+  with ``fieldTypes.ObjectId``.
+
+- The ``a`` field is an ``Int``, a data type available in Spark.
+
+Spark DataFrames
+~~~~~~~~~~~~~~~~
+
+To create DataFrames with custom MongoDB data types:
+
+- Create RDDs using custom MongoDB BSON types 
+  (e.g. ``ObjectId``). The MongoDB Spark Connector handles converting 
```

Comment:
This could probably be shorted to Spark Connector since you've introduced it using the full name earlier.